Assets/Scripts/R&D Tree/ResearchNode.cs

```diff
@@ -0,0 +1,106 @@
+using System.Runtime.CompilerServices;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+public class ResearchNode : MonoBehaviour
+{
+    [SerializeField] private RDTreeManager _treeManager;
+    public float ResearchValue { get; private set; }
+
+    public bool IsResearchInQueue { get; private set; } 
+
+    public ResearchNodeSetting NodeSetting {  get; private set; }
+
+    public ResearchStates CurrentResearchState { get;  private set; }
+
+    [HideInInspector]public UnityEvent ResearchDoneEvent;
+
+    private void Start()
+    {
+        NodeSetting = GetComponent<ResearchNodeSetting>();
+        _treeManager.AllNodes.Add(this);    
+    }   
+
+    /// <summary>
+    /// This function is called when the research is finished
+    /// </summary>
+    public void ResearchDone()
+    {
+        ResearchDoneEvent?.Invoke();
+        SetResearchBought();
+        SetNextStatesInTree();
+        RemoveResearchFromQueue();
+        _treeManager.ResearchFinished();
+    }
+
+    /// <summary>
+    /// This function is called to set the next nodes active when the research is done
+    /// </summary>
+    public void SetNextStatesInTree()
+    {
+        foreach (var connectedNode in NodeSetting.ConnectedResearchNodes)
+        {
+            if (connectedNode == null) 
+                continue;
+
+            connectedNode.CurrentResearchState = ResearchStates.available;
+
+            //Deleted this code when the visualisation of the tree is being made
+            connectedNode.GetComponent<TreeNodeDemo>().NodeStates();
+        }
+    }
+
+    /// <summary>
+    /// Call this function when you want the research to be developed
+    /// </summary>
+    /// <param name="value">The amount you want to add to the researchvalue
+    /// </param>
+    public void AddValue(float value)
+    {
+        if (CurrentResearchState != ResearchStates.inDevelopment) 
+            return;
+
+        ResearchValue += value;
+
+        if (ResearchValue >= NodeSetting.ResearchCompletionValue)
+            ResearchDone();
+    }
+    
+    public void SetResearchNotAvailable() => CurrentResearchState = ResearchStates.notAvailable;
```

Comment:
These functions are duplicated code and the naming is inconsistent. Change to something like `SetResearchStatus(ResearchState.notAvailable);`